### DISCUSSION TITLE
存储过程怎么强制指定在从库上执行

### DISCUSSION THREAD

**Discussion by Wolffy101** [dotnetcore/FreeSql#1833] [Q&A] [answered]
我有一个查询的存储过程 **PRO_XXX_GET**，但是名称并不是SELECT 开头,但是我想手动指定在从库上查询。我并没有找到这个方法,我看到代码判断从库执行的方法是一个静态方法，并不能重写
``` chsharp
 public static bool IsFromSlave(string cmdText)
 {
      return cmdText.StartsWith("SELECT ", StringComparison.CurrentCultureIgnoreCase) ||
          cmdText.StartsWith("WITH ", StringComparison.CurrentCultureIgnoreCase);
 }
```

**Comment by 2881099** [comment, marked as answer]
v3.2.831-preview20240625 

- 优化 Ado读写分离存储过程默认使用从库；

**Reply by 2881099** [reply]
已发布nuget，等5-10分钟后更新

**Reply by Wolffy101** [reply]
看到了修改是只要存储过程都是从库，但是我如果需要用主库要怎么怎么写

**Reply by 2881099** [reply]
只是查询api才会，例如 Ado.Query

Ado.ExecuteNoQuery 不灵活这个机制

**Reply by Wolffy101** [reply]
> 只是查询api才会，例如 Ado.Query
> 
> Ado.ExecuteNoQuery 不灵活这个机制

好的，谢谢
